perf/perf_nmem.py

```diff
@@ -0,0 +1,248 @@
+                                             ignore_status=True,
+                                             allow_output_check='combined')
+            if rc:
+                failed_events.append(self.all_events[key])
+        if failed_events:
+            self.fail("Failed with events: %s " % failed_events)
+
+    def test_mixed_events(self):
+        # If there are more than one PMU then try running mixed events
+        if len(self.pmu_list) > 1:
+            mix_events = []
+            for keys in self.all_events.keys():
+                mix_events.append(self.all_events[keys][0])
+            op = process.system_output("perf stat -e '{%s}' sleep 1"
+                                       % (",".join(mix_events)), shell=True,
+                                       ignore_status=True,
+                                       allow_output_check='combined')
+            er_ln = "The events in group usually have to be from the same PMU"
+            # Expecting failure with the string in 'er_ln'
+            if er_ln in op.decode():
+                self.log.info("Expected failure with mixed events")
+            else:
+                self.fail("Expected a failure but test pass.")
+        else:
+            self.cancel("With single PMU mixed events test is not possible.")
+
+    def _get_cpumask(self, event_type):
+        # Reading the cpumask and return
+        event_cpumask_file = "%s%s/cpumask" % (self.base_dir, event_type)
+        return int(genio.read_file(event_cpumask_file).rstrip('\t\r\n\0'))
+
+    def test_pmu_cpumask(self):
+        # Checking whether the PMU cpumask exists or not
+        failed_list = []
+        for pmu in self.pmu_list:
+            pmu_cpumask = self._get_cpumask(pmu)
+            if not str(pmu_cpumask):
+                failed_list.append(pmu_cpumask)
+            self.log.info("%s contains cpumask = %s" % (pmu, pmu_cpumask))
+        if failed_list:
+            self.fail("Not found cpumask file for %s" % failed_list)
+
+    def _check_cpumask(self):
+        pmu_cpumask = []
+        for pmu in list(self.all_events.keys()):
+            pmu_cpumask.append(self._get_cpumask(pmu))
+        if pmu_cpumask:
+            self.log.info("Found cpumasks are = %s" % pmu_cpumask)
+        else:
+            self.fail("Fail to get cpumask")
+        # Get the unique value from the cpumask list
+        if len(set(pmu_cpumask)) != 1:
+            self.fail("cpumask values are not same: %s" % pmu_cpumask)
+
+    def test_cpumask(self):
+        # Checking all the PMUs cpumask is same or not
+        self._check_cpumask()
+
+    def _cpu_on_off(self, cpu_number, disable_flag=True):
+        cpu_file = "/sys/bus/cpu/devices/cpu%s/online" % cpu_number
+        if disable_flag:
+            genio.write_one_line(cpu_file, "0")
+            self.log.info("Offline CPU: %s" % cpu_number)
+        else:
+            genio.write_one_line(cpu_file, "1")
+            self.log.info("Online CPU: %s" % cpu_number)
```

Comment:
using avocado CPU module will be good here as we already using CPU module

Reply:
done